### PR TITLE
Added CIELab perceptual color distance

### PR DESCRIPTION
**Pull request by notmahi**
Changed the color distance calculator for the pallete to CIELab perceptual color distance instead of Euclidean distance.

Testing: done locally with a browser; however with my limited tests it's really hard to tell if it improved anything.

### CODE REVIEW

File: main.js [hunk 1]
```diff
@@ -202,4 +199,88 @@ function hexToRgba(hex) {
...
+}
+
+/**
+ * Convert an sRGB color to CIELAB
+ * http://www.easyrgb.com/en/math.php
+ * 
+ * @param {Array} sRGB sRGB color in array format
+ */
+function sRGBToCIELab(sRGB) {
+  // Convert sRGB to CIELab using the math from
```
Comment (lahuang4) [inline main.js]: Unfinished sentence?

File: main.js [hunk 2]
```diff
@@ -202,4 +199,88 @@ function hexToRgba(hex) {
     parseInt(hex.slice(3, 5), 16),
     parseInt(hex.slice(5, 7), 16),
     255];
-}
\ No newline at end of file
+}
+
+/**
+ * Convert an sRGB color to CIELAB
+ * http://www.easyrgb.com/en/math.php
```
Comment (lahuang4) [inline main.js:225]: It's not clear to me what CIELAB is from this reference -- is there a better reference to link here?

File: main.js [hunk 3]
```diff
@@ -202,4 +199,88 @@ function hexToRgba(hex) {
...
+  var L = (116 * var_Y) - 16;
+  var a = 500 * (var_X - var_Y);
+  var b = 200 * (var_Y - var_Z);
+
+  return [L, a, b];
+}
+
+/**
+ * calculate the perceptual distance between colors in CIELAB
+ * https://github.com/THEjoezack/ColorMine/blob/master/ColorMine/ColorSpaces/Comparisons/Cie94Comparison.cs
```
Comment (lahuang4) [inline main.js]: This link doesn't work for me, please take another look at it.

File: main.js (hunk 2 above)
Comment (lahuang4) [reply main.js:225]: Would also appreciate a reference to what XYZ and LAB are, thanks.

File: main.js (hunk 3 above)
Comment (notmahi) [reply main.js]: Added a different algorithm with the source for the math.

File: main.js (hunk 2 above)
Comment (notmahi) [reply main.js:225]: Added a link to an easy explanation and wikipedia articles for all the new terminologies.

File: main.js (hunk 1 above)
Comment (notmahi) [reply main.js]: Resolved in latest commit, sorry!

File: main.js [hunk 4]
```diff
@@ -196,13 +196,28 @@ function loadImageWithPalette() {
 
 function hexToRgba(hex) {
   return [parseInt(hex.slice(1, 3), 16),
-    parseInt(hex.slice(3, 5), 16),
-    parseInt(hex.slice(5, 7), 16),
+  parseInt(hex.slice(3, 5), 16),
+  parseInt(hex.slice(5, 7), 16),
```
Comment (lahuang4) [inline main.js]: Accidental unindent?

File: main.js [hunk 5]
```diff
@@ -196,25 +196,41 @@ function loadImageWithPalette() {
...
     255];
 }
```
Comment (lahuang4) [inline main.js:203]: Can you add an extra comment separator here:

//
// Color distance helpers
//

**Review by lahuang4 — COMMENTED** [review]
Thanks for the much-improved comments and references!

File: main.js (hunk 5 above)
Comment (notmahi) [reply main.js:203]: Done!

**Pull request merged**